backend/spec/models/enrollment_spec.rb

```diff
@@ -21,6 +21,26 @@
     end
   end
 
+  describe "validations" do
+    let(:enrollment_creator) { create(:user, organization_kind: :clamart) }
+    let(:enrollment) {
+      create(:enrollment, :franceconnect, status: "validated", organization_kind: :clamart, user: enrollment_creator)
+    }
+
+    it "can't create 2 copies for the same enrollment" do
+      create(:enrollment, :franceconnect, status: "validated", copied_from_enrollment_id: enrollment.id, organization_kind: :clamart, user: enrollment_creator)
+      new_copy = build(:enrollment, :franceconnect, status: "draft", copied_from_enrollment_id: enrollment.id, organization_kind: :clamart, user: enrollment_creator)
+
+      expect(new_copy).not_to be_valid
```

Comment:
Techniquement ce n'est pas une `copy` que tu fais ici, il faudrait utiliser la méthode `copy`.

```suggestion
      create(:enrollment, :franceconnect, status: "validated", copied_from_enrollment_id: enrollment.id, organization_kind: :clamart, user: enrollment_creator)
      expect(enrollment.copy(user)).not_to be_valid
```

idem plus bas.

Reply:
Rectifié 👍🏻